pkg/handlers/generic/lifecycle/utils/utils.go

```diff
@@ -86,3 +126,113 @@ func EnsureNamespace(ctx context.Context, c ctrlclient.Client, name string) erro
 
 	return nil
 }
+
+func RetrieveValuesTemplateConfigMap(
+	ctx context.Context,
+	c ctrlclient.Client,
+	configMapName,
+	defaultsNamespace string,
+) (*corev1.ConfigMap, error) {
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: defaultsNamespace,
+			Name:      configMapName,
+		},
+	}
+	configMapObjName := ctrlclient.ObjectKeyFromObject(
+		configMap,
+	)
+	err := c.Get(ctx, configMapObjName, configMap)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"failed to retrieve installation values template ConfigMap %q: %w",
+			configMapObjName,
+			err,
+		)
+	}
+	return configMap, nil
+}
+
+func CreateStorageClass(
+	storageConfig v1alpha1.StorageClassConfig,
+	defaultsNamespace string,
+	provisionerName v1alpha1.StorageProvisioner,
+	isDefault bool,
+) *storagev1.StorageClass {
+	var volumeBindingMode *storagev1.VolumeBindingMode
+	switch storageConfig.VolumeBindingMode {
+	case v1alpha1.VolumeBindingImmediate:
+		volumeBindingMode = ptr.To(storagev1.VolumeBindingImmediate)
+	case v1alpha1.VolumeBindingWaitForFirstConsumer:
+		volumeBindingMode = ptr.To(storagev1.VolumeBindingWaitForFirstConsumer)
+	}
+	var reclaimPolicy *corev1.PersistentVolumeReclaimPolicy
+	switch storageConfig.ReclaimPolicy {
+	case v1alpha1.VolumeReclaimRecycle:
+		reclaimPolicy = ptr.To(corev1.PersistentVolumeReclaimRecycle)
+	case v1alpha1.VolumeReclaimDelete:
+		reclaimPolicy = ptr.To(corev1.PersistentVolumeReclaimDelete)
+	case v1alpha1.VolumeReclaimRetain:
+		reclaimPolicy = ptr.To(corev1.PersistentVolumeReclaimRetain)
+	}
+	var params map[string]string
```

Comment:
I think we should also set the params for nutanix https://github.com/nutanix/helm/blob/4ce73556497c851d1e17f197ee96f99b773c49f8/charts/nutanix-csi-storage/templates/ntnx-sc.yaml#L23-L28

The user is not deploying the CSI driver or creating this Secret, so they wouldn't really know what values to put here.

Reply:
The docs that i got this from: https://portal.nutanix.com/page/documents/details?targetId=CSI-Volume-Driver-v2_6:csi-csi-plugin-manage-nfs-storage-t.html